go.mod

```diff
@@ -69,29 +71,33 @@ replace (
-	k8s.io/api => k8s.io/api v0.21.0-rc.0
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.21.0-rc.0
-	k8s.io/apimachinery => k8s.io/apimachinery v0.21.0-rc.0
-	k8s.io/apiserver => k8s.io/apiserver v0.21.0-rc.0
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.21.0-rc.0
-	k8s.io/client-go => k8s.io/client-go v0.21.0-rc.0
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.21.0-rc.0
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.21.0-rc.0
-	k8s.io/code-generator => k8s.io/code-generator v0.21.0-rc.0
-	k8s.io/component-base => k8s.io/component-base v0.21.0-rc.0
+	k8s.io/api => k8s.io/api v0.21.3
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.21.3
+	k8s.io/apimachinery => k8s.io/apimachinery v0.21.3
+	k8s.io/apiserver => k8s.io/apiserver v0.21.3
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.21.3
+	k8s.io/client-go => k8s.io/client-go v0.21.3
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.21.3
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.21.3
+	k8s.io/code-generator => k8s.io/code-generator v0.21.3
+	k8s.io/component-base => k8s.io/component-base v0.21.3
 	k8s.io/component-helpers => k8s.io/component-helpers v0.20.0-alpha.2.0.20210708095128-55a3896515e9
 	k8s.io/controller-manager => k8s.io/controller-manager v0.20.0-alpha.1.0.20210712075914-83508d18fce2
-	k8s.io/cri-api => k8s.io/cri-api v0.21.0-rc.0
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.21.0-rc.0
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.21.0-rc.0
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.21.0-rc.0
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.21.0-rc.0
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.21.0-rc.0
-	k8s.io/kubectl => k8s.io/kubectl v0.21.0-rc.0
-	k8s.io/kubelet => k8s.io/kubelet v0.21.0-rc.0
-	k8s.io/kubernetes => k8s.io/kubernetes v1.21.0-rc.0
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.21.0-rc.0
-	k8s.io/metrics => k8s.io/metrics v0.21.0-rc.0
+	k8s.io/cri-api => k8s.io/cri-api v0.21.3
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.21.3
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.21.3
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.21.3
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.21.3
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.21.3
+	k8s.io/kubectl => k8s.io/kubectl v0.21.3
+	k8s.io/kubelet => k8s.io/kubelet v0.21.3
+	k8s.io/kubernetes => k8s.io/kubernetes v1.21.3
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.21.3
+	k8s.io/metrics => k8s.io/metrics v0.21.3
 	k8s.io/mount-utils => k8s.io/mount-utils v0.23.0-alpha.0
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.21.0-rc.0
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.21.3
 )
+
+// Workaround to deal with https://github.com/kubernetes/klog/issues/253
+// Should be deleted when https://github.com/kubernetes/klog/pull/242 is merged
+exclude github.com/go-logr/logr v1.0.0
```

Comment:
it worked for me adding it as a `replace`

```
github.com/go-logr/logr =>  github.com/go-logr/logr v0.4.0
```

Reply:
The "official" recommendation is to use `exclude` https://github.com/kubernetes/klog/issues/253#issuecomment-883465311 but I can try with replace.

Reply:
done

Reply:
I trust dims more than me, just use the one you want meanwhile unblocks you :smile: